pkg/clients/database/cockroach/cockroach_test.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cockroach
+
+import (
+	"testing"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/google/go-cmp/cmp"
+	rookv1alpha1 "github.com/rook/rook/pkg/apis/cockroachdb.rook.io/v1alpha1"
+	rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplaneio/stack-rook/apis/database/v1alpha1"
+	corev1alpha1 "github.com/crossplaneio/stack-rook/apis/v1alpha1"
+)
+
+const (
+	name      = "cool-name"
+	namespace = "cool-namespace"
+
+	providerName         = "cool-rook"
+	connectionSecretName = "cool-connection-secret"
+)
+
+type cockroachClusterModifier func(*v1alpha1.CockroachCluster)
+
+func withCockroachNodeCount(c int) cockroachClusterModifier {
+	return func(i *v1alpha1.CockroachCluster) { i.Spec.CockroachClusterParameters.Storage.NodeCount = c }
+}
+
+func cockroachCluster(im ...cockroachClusterModifier) *v1alpha1.CockroachCluster {
+	i := &v1alpha1.CockroachCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.CockroachClusterSpec{
+			ResourceSpec: runtimev1alpha1.ResourceSpec{
+				ProviderReference:                &corev1.ObjectReference{Name: providerName},
+				WriteConnectionSecretToReference: &runtimev1alpha1.SecretReference{Name: connectionSecretName},
+			},
+			CockroachClusterParameters: v1alpha1.CockroachClusterParameters{
+				Name:        name,
+				Namespace:   namespace,
+				Annotations: corev1alpha1.Annotations(map[string]string{"label": "value"}),
+				Storage: corev1alpha1.StorageScopeSpec{
+					NodeCount: 3,
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "rook-cockroachdb-test",
+							},
+							Spec: corev1.PersistentVolumeClaimSpec{
+								AccessModes: []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"},
+								// Test does not check resource requirements due cmp pkg
+								// inability to compare unexported fields.
```

Comment:
https://godoc.org/github.com/google/go-cmp/cmp#AllowUnexported

You may already be aware, but you _can_ compare unexported fields with cmp. It is considered bad practice though, especially for types that are outside your control.